src/main/java/org/kitteh/irc/client/library/feature/InteractionDefaults.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.kitteh.irc.client.library.feature;
+
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+public class InteractionDefaults implements InteractionMap {
+    protected Map<Interaction, String> defaults = new HashMap<>();
+    protected final String defaultString;
+
+    /**
+     * @param defaultString Provide a default string to use for when a key is not set
+     */
+    public InteractionDefaults(@Nullable String defaultString) {
+        if (defaultString != null && defaultString.isEmpty()) {
+            // Both "" and null are legal here, technically. But for KICL to perform best (and to avoid != null && string.isEmpty() checks),
+            // we enforce a requirement on null.
+            throw new IllegalArgumentException("defaultString may not be an empty string, use null instead.");
+        }
+        this.defaultString = defaultString;
+    }
+
+    public InteractionDefaults() {
+        this.defaultString = null;
+    }
+
+    /**
+     * This method will throw a RuntimeException if you try to define a value that
+     * is already defined. This often indicates a problem and #replaceDefault should be used
+     * instead.
+     *
+     * @param key The interaction to set a value for
+     * @param defaultString The default value to assume when nothing is provided for this Interaction
+     * @return Returns this to permit chaining
+     */
+    public InteractionDefaults setDefault(Interaction key, @Nullable String defaultString) {
+        if (this.defaults.get(key) != null) {
+            throw new RuntimeException("The default value for " + key.toString() + " is already set with value: " + defaults.get(key));
+        }
+
+        this.defaults.put(key, defaultString);
+
+        return this;
+    }
+
+    /**
+     *
+     * @param key The default Interaction key to obtain the default of
+     * @return The string of the default value, or the predefined defaultString
+     */
+    public String getDefault(Interaction key) {
+        String value = this.defaults.get(key);
+
+        if (value == null) {
+            return this.defaultString;
+        }
+
+        return value;
+    }
+
+    /**
+     * @return This method returns a Map with your defaults, and will fill all Interaction keys that were not already
```

Comment:
Missing description. `@return` is a piece of it. Also the column count is off here too, as well as indentation.